### PR TITLE
feat: move to ts

### LINKED ISSUE #82: move all files to typescript

### PR DESCRIPTION
**Pull request by LinaYahya**
- moving js to  ts
- eslint configurations and fix linting problems

closes #82

### CODE REVIEW

File: src/config/hooks.ts [hunk 1]
```diff
@@ -106,21 +136,20 @@ export const useModelsSearch = (queryParams = {}) =>
         },
         {
           addQueryPrefix: true,
-        }
+        },
       );
       // cannot use axios https://github.com/miragejs/miragejs/issues/1006
       const response = await fetch(`${MODEL_SEARCH_ENDPOINT}${queryString}`);
```
Comment (spaenleh) [inline src/config/hooks.ts:142]: You can build the url using `new URL` and `URLSearchParams`. I personally find it easier than using qs.
```ts
const url = new URL(MODEL_SEARCH_ENDPOINT);
const query = new URLSearchParams({ type: 'models', q: queryParams.q ?? DEFAULT_QUERY });
url.search = query
const response = await fetch(url.toString())
```

File: src/data/db.ts [hunk 2]
```diff
@@ -56,6 +56,7 @@ const mockItem: AppItemType = {
   createdAt: new Date().toISOString(),
   updatedAt: new Date().toISOString(),
   type: ItemType.APP,
+  lang: 'en',
```
Comment (spaenleh) [inline src/data/db.ts:59]: Use constant defined in i18n config file

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good, just a few comments.  💪 

Great work ! let me know if you have questions

File: cypress/fixtures/item.ts [hunk 3]
```diff
@@ -1,4 +1,8 @@
-import { Item } from '@graasp/sdk';
+import { AppItemFactory } from '@graasp/sdk';
 
 export const MOCK_ITEM_ID = '1234-1234-123456-8123-123456';
-export const MOCK_SERVER_ITEM = { id: MOCK_ITEM_ID } as Item;
+export const MOCK_SERVER_ITEM = { id: MOCK_ITEM_ID };
+
+export const MOCK_APP_ITEM = AppItemFactory({
+  name: 'sketchfab item',
+});
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply cypress/fixtures/item.ts:8]: It's used inside model `AppSettings` `/cypress/fixtures/appSettings.ts`

File: src/components/views/builder/BuilderView.tsx [hunk 4]
```diff
@@ -36,9 +37,9 @@ const StyledFab = styled(Fab)(({ theme }) => ({
 const BuilderView = (): JSX.Element => {
   const { t } = useTranslation();
   const [open, setOpen] = useState(false);
-  const [search, setSearch] = useState<string | null>(null);
+  const [search, setSearch] = useState<string>(DEFAULT_QUERY);
```
Comment (spaenleh) [inline src/components/views/builder/BuilderView.tsx:40]: nice !

**Review by spaenleh — APPROVED** [review]
Very nice ! 🌷

**Pull request merged**